src/components/CollectionsAndPalettes/common/ItemCard/ItemCard.tsx

```diff
@@ -1,43 +1,111 @@
 import * as React from "react";
 import cn from "classnames";
 import styles from "../../common/ItemCard/ItemCard.module.scss";
+import { CollectionItemT } from "../../../../constants/types";
+import { AppContext, AppContextState } from "../../../../context/AppContext";
+import { get } from "lodash-es";
 
 interface Props {
   mode: any;
-  item: {
-    type: any;
-    title1: string;
-    title2: string;
-    title3: string;
-    price: string;
-    imagePath: string;
-  };
+  item: CollectionItemT;
 }
 
 export default class ItemCard extends React.Component<Props, any> {
-  renderImageItem(type) {
+  static contextType = AppContext;
+  context!: AppContextState;
+  materialItem: any;
+
+  mapAlgoleaToObject(): any {
+    let item = null;
+    if (this.props.item.material) {
+      item = this.props.item.material.id.toString();
+    } else {
+      item = this.props.item.upload;
+    }
+    const algoliaProduct = this.context.productsCache.getProduct(item);
+    const color = get(
+      algoliaProduct,
+      "data.color",
+      "this.props.item.material.color"
```

Comment:
The final param is the default value in case `algoliaProduct.data.color` is missing, and that will be displayed in case the product doesn't have the field or if we don't yet have the algolia products loaded. I think we should just make this be `material.color`